### PR TITLE
[zero-copy] Implement missing primitives

### PR DESCRIPTION
**Pull request by CraftSpider**
May need more testing, but these should be all the primitives in main but not zero-copy except the deprecated `seq`

### CODE REVIEW

File: src/zero_copy/primitive.rs [hunk 1]
```diff
@@ -318,6 +318,159 @@ pub fn any<I: Input + ?Sized, E: Error<I::Token>>() -> Any<I, E> {
...
+
+    fn go<M: Mode>(&self, inp: &mut InputRef<'a, '_, I, E, S>) -> PResult<M, Self::Output, E> {
+        let mut output = M::bind(|| C::default());
+
+        loop {
+            let start = inp.save();
+            match self.until.go::<M>(inp) {
+                Ok(out) => break Ok(M::combine(output, out, |output, out| (output, out))),
+                Err(_) => ()
+            }
```
Comment (zesterer) [inline src/zero_copy/primitive.rs:429]: `if let Ok(out) = ...`?

Comment (CraftSpider) [reply src/zero_copy/primitive.rs:429]: You've caught my most frequent mistake - When I refactor a match and end up with just an if-let, but never change it over. Fixing now

**Comment by wackbyte** [comment]
Can you add them to the prelude as well?

**Review by wackbyte — APPROVED** [review]

**Comment by zesterer** [comment]
Thanks for the PR!
